lldb/include/lldb/API/SBProcess.h

```diff
@@ -158,7 +158,7 @@ class LLDB_API SBProcess {
 
   lldb::SBError Destroy();
 
-  lldb::SBError Continue();
+  lldb::SBError Continue(RunDirection direction = RunDirection::eRunForward);
```

Comment:
Our public API has rules:
- can't change any existing API call, you can add an overload, but you can't change any public APIs that are already there. Anything in the `lldb` namespace can't be changed. 
- no virtual functions
- can't change any ivars or the size of the object

That being said, I would rather have a:
```
lldb::SBError ReverseContinue();
```
call than have everyone using the `Continue` API to say wether they want to go forward or in reverse.

Reply:
Yeah, I originally wrote everything as you suggested, and @jimingham asked me to change it to what you see now. So I need you two to come to an agreement and tell me what it is.

Reply:
I think it should be possible for most of lldb not to care which direction we happen to be going in.  We just need to know the current direction, or the intended direction of the active thread plan.  If we have to start putting in lots of special purpose logic for forward and backward outside of the ProcessGDBRemote plugin, I think we're doing it wrong.
But that's going to be harder to do if we start having a bunch of forward and reverse specific functions, since they all act as sites where you forget the "current direction" and then have to re-infer it later on.  It's particularly awkward having some take a direction and some be special purpose forward or backward.

Reply:
@jimingham: do you want all process control function to remain the same and then add a direction control? Something like:
```
lldb::SBError SetRunDirection(RunDirection direction);
```
Jim is this what you are thinking?


And my API comments stand though, we can't change a pre-existing API. We can add new overloads though

Reply:
I have made this change.